docs/testing/runtimes.md

```diff
@@ -0,0 +1,117 @@
+
+- Store shared data in an online database that is accessible to all the runtimes.
+- Store shared data in a cookie for the add-in's domain to share it between browser runtimes. JavaScript-only runtimes do not support cookies.
+
+For more information, see [Persist add-in state and settings](../develop/persisting-add-in-state-and-settings.md) and [Manage state and settings for an Outlook add-in](../outlook/manage-state-and-settings-outlook.md).
+
+## JavaScript-only runtime
+
+The JavaScript-only runtime that is used in Office add-ins is a modification of an open source runtime originally created for [React Native](https://reactnative.dev/). It contains a JavaScript engine supplemented with support for [WebSockets](https://developer.mozilla.org/docs/Web/API/WebSockets_API), [Full CORS (Cross-Origin Resource Sharing)](https://developer.mozilla.org/docs/Web/HTTP/CORS), and [local storage](https://developer.mozilla.org/docs/Web/API/Window/localStorage). It doesn't have a rendering engine, and it doesn't support cookies.
+
+This type of runtime is used in Outlook event-based (autolaunch) tasks and in Excel custom functions *except* when the custom function is [sharing a runtime](#shared-runtime). 
+
+- When used for an Excel custom function, the runtime starts up when either the worksheet recalculates or the custom function calculates. It doesn't shut down until the workbook is closed.  
+- When used in an Outlook event-based task, the runtime starts up when the event occurs. It ends when the first of the following occurs:
+
+  - The event handler calls the `completed` method of its event parameter.
+  - 5 minutes has elapsed since the triggering event.
+  - The user changes focus from the window where the event was triggered, such as a message compose window.
+
+## Browser runtime
+
+Office add-ins use a different browser type runtime depending on the platform in which Office is running (web, Mac, or Windows), and on the version and build of Windows and Office. For example, if the user is running Office on the web in a FireFox browser, then the Firefox runtime is used. If the user is running Office on Mac, then the Safari runtime is used. If the user is running Office on Windows, then either an Edge or Internet Explorer provides the runtime, depending on the version of Windows and Office. Details can be found in [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
+
+All of these runtimes include an HTML rendering engine and provide support for [WebSockets](https://developer.mozilla.org/docs/Web/API/WebSockets_API), [Full CORS (Cross-Origin Resource Sharing)](https://developer.mozilla.org/docs/Web/HTTP/CORS), and [local storage](https://developer.mozilla.org/docs/Web/API/Window/localStorage), and cookies. 
+
+A browser runtime lifespan varies depending on the feature that it implements.
+
+- When an add-in with a task pane is launched a browser runtime starts. It shuts down when the add-in is closed.
```

Comment:
should we say It shuts down when the task pane is closed?

Reply:
Good catch. If the runtime isn't shared, it shuts down when the task pane is closed.

Reply:
fixed